imageio/plugins/pillow.py

```diff
@@ -666,9 +666,9 @@ def ndarray_to_pil(arr, format_str=None):
         mode = "L"
         mode_base = "L"
 
-    if mode == "I;16":
-        # Image.fromarray doesn't seem to be compatible with 'I;16' formats
-        # Therefore, we fallback to this expensive workaround.
+    if mode == "I;16" and int(getattr(Image, "__version__", "0")[0]) < 6:
```

Comment:
We could also fiddle a `.split(".")` in there so this continues to work with Pillow 10.x. What do you think?

Reply:
Thank you for spotting that, this is why having a second set of eyes is so important (and why programming while in bed is perhaps not my best habit).

The expression is already getting rather long/convoluted; would you prefer that I break off the version check into a separate if statement? A la:
```python
if mode == "I;16":
    pillow_version = getattr(Image, "__version__", "0")
    if pillow_version.split(".")[0] >= 6:
        return Image.fromarray(arr, mode)
    array_buffer = arr.tobytes()
    if arr.ndim == 2:
        im = Image.new(mode_base, arr.T.shape)
        im.frombytes(array_buffer, "raw", mode)
    else:
        image_shape = (arr.shape[1], arr.shape[0])
        im = Image.frombytes(mode, image_shape, array_buffer)
    return im
else:
    return Image.fromarray(arr, mode)
```
Some inherent pros and cons to this approach compared with sticking on the split. Any preference?

Reply:
I don't have a strong preference, but if the line gets too long the style checker will complain, and then splitting it over multiple lines would indeed be better :)

Reply:
Decided to keep the multi-digit major version fix inline as it still fits comfortably on a single line and it avoids duplicating the default return.